### DISCUSSION TITLE
no data after I put Api-Key inside the env.local

### DISCUSSION THREAD

**Discussion by NikCrmr** [neuefische/questions#826] [Web] [answered]
After i moved the Api-Key inside the .env.local I don't receive any data from my Api anymore.

in the .env.local
`X_RAPID_API_KEY=MY-KEY`

in my page-file

` 
   const options = {
      method: "POST",
      headers: {
        "content-type": "application/json",
        Accept: "application/json",
        "X-RapidAPI-Key": process.env.X_RAPID_API_KEY,
        "X-RapidAPI-Host": "ekman-emotion-analysis.p.rapidapi.com",
      },
      body: JSON.stringify([
        {
          id: "1",
          language: language,
          text: inputText,
        },
      ]),
    };
`
   

The .env.local is already used for the mongodb database and the github OAuth and it works there.

**Comment by onemanwenttomow** [comment]
It depends. Can you add a link to your repo?

It could be that you are trying to make the request from the client side. 

If you want to use env variables on the client side (i.e. the browser) then you need to add [NEXT_PUBLIC_ to the front](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser). 

Please really think about if this is what you want though. This will expose the X_RAPID_API key to the client (i.e. compromise it). 

If you don't want it to be available then you will need to move the api routes and make the request from there (like the db queries).

**Reply by NikCrmr** [reply]
https://github.com/MathiSans/capstone_mood-tracker/pull/92
here is the Pull request to the feature and here the branch code:
https://github.com/MathiSans/capstone_mood-tracker/tree/ekmann-emotion-text-analyse

**Reply by onemanwenttomow** [reply, marked as answer]
So it looks like the request to `https://ekman-emotion-analysis.p.rapidapi.com/ekman-emotion?all=true` using the `X-RapidAPI-Key` is currently made on the **client side**. I.e. in `handleSubmit`. 

If this key needs to remain a secret (i.e. it is tied to a rate limit or credit card) then you will want to move this request to the **server**.

In nextjs that is our api routes. 

So handleSubmit will need to make a request to an api route (i.e. /api/something)
That api route will import the secrets (i.e. process `process.env.X_RAPID_API_KEY`) and make the request. Once you get a response you send it back to the client and then handle that in `handleSubmit`.

**Reply by NikCrmr** [reply]
Hey Pete! One question remains. When I push the branch to vercel, do I also have to put the API key inside of Vercel environment variable? Vercel doesn't have my .env.local cause it's in the gitignore list.

**Reply by onemanwenttomow** [reply]
Exactly!

There is a [nice article here](https://vercel.com/docs/projects/environment-variables).

You won't need to change anything in the code, just add them to Vercel.
